jdaviz/core/launcher.py

```diff
@@ -74,59 +82,91 @@ def _launch_config_with_data(config, data=None, show=True, **kwargs):
+        self.compatible_configs = configs
+
+        self.config_icons = {
+            'cubeviz': self.cubeviz_icon,
+            'specviz': self.specviz_icon,
+            'specviz2d': self.specviz2d_icon,
+            'mosviz': self.mosviz_icon,
+            'imviz': self.imviz_icon
+        }
+
+        start_path = os.environ.get('JDAVIZ_START_DIR', os.path.curdir)
+        self._file_chooser = FileChooser(start_path)
+        self.components = {'g-file-import': self._file_chooser}
+        self.loaded_data = None
+        self.hint = "No filepath provided. Choose a config to load a blank tool:"
+        super().__init__(*args, **kwargs)
+
+    @observe('filepath')
+    def _filepath_changed(self, *args):
+        self.hint = "Please wait. Identifying file..."
+        self.compatible_configs = []
+        if self.filepath in (None, ''):
+            self.compatible_configs = self.configs
+            self.loaded_data = None
+            self.hint = "No filepath provided. Choose a config to load a blank tool:"
         else:
-            compatible_helpers, loaded_data = identify_helper(filepath)
-            if len(compatible_helpers) > 0 and loaded_data is None:
-                loaded_data = filepath
-
-        for config, btn in btns.items():
-            btn.disabled = not (config in compatible_helpers)
-
-    id_data_btn = v.Btn(class_="ma-2", outlined=True, color="primary",
-                        children=[v.Icon(children=["mdi-magnify"])])
-    id_data_btn.on_event('click', lambda btn, event, data: enable_compatible_configs(btn.value))
-    jslink((text_field, 'v_model'), (id_data_btn, 'value'))
-
-    filepath_row.children = [text_field, id_data_btn]
-
-    # Create Launcher
-    main.children = [intro_row, filepath_row, btn_row]
+            path = Path(self.filepath)
+            if not path.is_file():
+                self.loaded_data = None
+                self.hint = "Cannot find file on disk. Please check your filepath"
+            else:
+                try:
+                    self.compatible_configs, self.loaded_data = identify_helper(self.filepath)
+                    self.hint = "Compatible configs identified! Please select your config:"
+                except Exception:
+                    self.hint = "Could not autoidentify compatible config. Please manually select config to load your data into:"
+                    self.compatible_configs = self.configs
+                    self.loaded_data = self.filepath
+                finally:
+                    if len(self.compatible_configs) > 0 and self.loaded_data is None:
+                        self.loaded_data = self.filepath
+        # Clear hint if it's still stuck on "Identifying". We're in an ambiguous state
+        self.hint = '' if self.hint == "Please wait. Identifying file..." else self.hint
+
+    def vue_choose_file(self, *args, **kwargs):
+        print(self._file_chooser.file_path)
```

Comment:
```suggestion
```

Reply:
Done in https://github.com/spacetelescope/jdaviz/pull/2311/commits/81f2025d82528e21d0a0bb5c6f9eb8279f47ae02